packages/cron/src/jobs/pins.js

```diff
@@ -52,21 +53,29 @@ export async function updatePinStatuses ({ cluster, db }) {
       const { pin } = req
       let peerMap
 
+      // Get status of pin or check later
       try {
         peerMap = await getPinStatus(pin.contentCid)
       } catch (err) {
         reSyncPins.push(pin)
         return null // Cluster could not find the content, please check later
       }
 
+      // Is this pin tracked by our cluster?
       if (!peerMap[pin.location.peerId]) {
         return null // not tracked by our cluster
       }
 
       const status = toPinStatusEnum(peerMap[pin.location.peerId].status)
 
       if (status !== 'Pinned' && status !== 'Remote') {
-        reSyncPins.push(pin)
+        const cid = status === 'Unpinned' ? downgradeCid(pin.contentCid) : pin.contentCid
+        try {
+          peerMap = await getPinStatus(cid)
+        } catch (err) {
+          reSyncPins.push(pin)
+          return null
+        }
```

Comment:
We end up unnecessarily calling `getPinStatus` twice in the case where status is not unpinned, pinned or remote. We also don't push onto `reSyncPins` unless there is an error calling `getPinStatus`.

We need to do this check after the first call to `getPinStatus` _if_ the status is `Unpinned` and then continue as before.

i.e.

```js
      try {
        peerMap = await getPinStatus(pin.contentCid)
      } catch (err) {
        reSyncPins.push(pin)
        return null // Cluster could not find the content, please check later
      }

      if (!peerMap[pin.location.peerId]) {
        return null // not tracked by our cluster
      }

      let status = toPinStatusEnum(peerMap[pin.location.peerId].status)

      // If "Unpinned", try a v0 CID
      if (status === 'Unpinned') {
        let cidv0
        try {
          cidv0 = downgradeCid(pin.contentCid)
        } catch (err) {
          // Cluster says this is unpinned, but it is not a v0 CID.
          log(`⚠️ Unable to downgrade CID: ${pin.contentCid}`)
          reSyncPins.push(pin)
          return null
        }

        try {
          peerMap = await getPinStatus(cidv0)
        } catch (err) {
          reSyncPins.push(pin)
          return null
        }

        if (!peerMap[pin.location.peerId]) {
          return null // not tracked by our cluster
        }

        status = toPinStatusEnum(peerMap[pin.location.peerId].status)
      }

      if (status !== 'Pinned' && status !== 'Remote') {
        // as before...
```